Sources/VideoOptionsControlBar.swift

```diff
@@ -0,0 +1,147 @@
+/*****************************************************************************
+ * VideoOptionsControlBar.swift
+ *
+ * Copyright © 2019 VLC authors and VideoLAN
+ *
+ * Authors: Robert Gordon <robwaynegordon@gmail.com>
+ *
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+@objc (VLCVideoOptionsControlBarDelegate)
+protocol VideoOptionsControlBarDelegate: class {
+    func didToggleFullScreen(_ optionsBar: VideoOptionsControlBar)
+    func didToggleRepeat(_ optionsBar: VideoOptionsControlBar)
+    func didSelectSubtitle(_ optionsBar: VideoOptionsControlBar)
+    func didSelectMoreOptions(_ optionsBar: VideoOptionsControlBar)
+    func didToggleInterfaceLock(_ optionsBar: VideoOptionsControlBar)
+}
+
+@objc (VLCVideoOptionsControlBar)
+class VideoOptionsControlBar: UIStackView {
+    
+    // MARK: Instance variables
+    @objc weak var delegate: VideoOptionsControlBarDelegate?
+    private var rptMode: VLCRepeatMode = .doNotRepeat
+    
+    @objc var orientationAxis: NSLayoutConstraint.Axis {
+        set {
+            // rotate the control bar's orientation by switching it's height and width values and changing it's layout positioning
+            axis = newValue
+            let minX: CGFloat = frame.maxX - frame.size.height
+            let minY: CGFloat = frame.maxY - frame.size.width
+            frame = CGRect(x: minX, y: minY, width: frame.size.height, height:frame.size.width)
+        }
+        
+        get {
+            return axis
+        }
+    }
```

Comment:
if we change to constraints I think we will not need this part though

Reply:
I decided to just leave this in for now because the next phase would be to add the constraints and then I can just replace it with the intended constraints.